### PR TITLE
LGA-2602 and LGA-2601 Add test to edit financial eligibility

### PR DESCRIPTION
**Pull request by jogilder**
## What does this pull request do?

Bug showed that a specialist provider could not edit the eligibility check. These new tests check to see if the provider and the operator can edit and save changes to an eligibility check. 

There is a backend PR to create the cases via fixtures.

There are also changes to allow use of chromedriver
 
## Any other changes that would benefit highlighting?

The two edit case tests (one operator, one specialist provider) use the fixtures from this [PR](https://github.com/ministryofjustice/cla_backend/pull/976). The tests pass locally when pointing to those images.

However, the downside to using fixtures is that after these tests have run once locally, they will either fail immediately after. In the case of our new tests, if the same case is being accessed and amended with the same figures over and over again it could lead to a false positive. We could reset this at the end of a test, we could also consider randomising what is entered in these number fields, saving the value in the context, and asserting it the same in the subsequent step. 

The test operator case list and specialist provider case list have different html structures. So we were not able to reuse steps across these two. Some of these steps are within the frontend_specialist_provider.py file. It may be worth splitting the test operator ones out to reduce confusion.

## Checklist

- [x] Provided JIRA ticket number in the title, e.g. "LGA-152: Sample title"

### CODE REVIEW

**Comment by said-moj** [comment]
@jogilder have you tried using https://pypi.org/project/chromedriver-py/

It has a method for auto locating the chromedriver so you don't have to look it up

**Comment by sarahyjja** [comment]
1. @means_test_operator_edit_case is the test to be pursue
2. Some @todos are guiding the next person for implementation of what's left
3. This doesn’t have to be finish if the next person want to go directly to point **4.**
4. The goal would be to create a fixture for the test_operator and the specialist_provider
=> To do so, create the fixtures in `cla_backend` repo (take eg. to `legalaid>fixtures`)
=> Add them in `create_db.sh>load_end_to_end_test_data()`
=> Refer the Reference Number attributed in `backend` in a variable in the `E2E` repo
=> Bring it inside of your test (take eg. to `constant.py>CLA_SPECIALIST_CASE_TO_ACCEPT`)

File: behave/features/steps/frontend_specialist_provider.py [hunk 1]
```diff
@@ -53,31 +57,38 @@ def step_impl_view_client_details(context):
     compare_client_details_with_backend(context, case_id, client_section)
 
 
-@step("I select a case to accept from the dashboard")
-def step_select_special_provider_case(context):
-    case_reference = CLA_SPECIALIST_CASE_TO_ACCEPT
-    check_only_unaccepted_cases = True
-    select_a_case(context, case_reference, check_only_unaccepted_cases)
+@step('I select the "Assign manually" button')
```
Comment (MishNajam) [inline behave/features/steps/frontend_specialist_provider.py:60]: Check if using this anywhere in test scenarios

Comment (MishNajam) [reply behave/features/steps/frontend_specialist_provider.py:60]: Deleted when replacing set up steps for the fixtures

File: behave/helper/constants.py [hunk 2]
```diff
@@ -129,6 +131,20 @@
         "media_code": "Don't Know",
         "source": "Phone",
     },
+    "EDIT": {
```
Comment (MishNajam) [inline behave/helper/constants.py:134]: delete

File: behave/features/steps/frontend_specialist_provider.py [hunk 3]
```diff
@@ -541,3 +552,30 @@ def step_impl_csv_error_details(context):
...
+    for row in context.table:
+        label = row["question"]
+        value = row["answer"]
+        label_format = label.ljust(len(label) + 1)
+        context.helperfunc.find_by_xpath(
+            f"//span[contains(text(),'{label_format}')]/../input"
+        ).send_keys(value)
+
+
+@step("I have created a case to edit in hours")
```
Comment (MishNajam) [inline behave/features/steps/frontend_specialist_provider.py:568]: delete this and the step below

File: behave/features/cla_frontend/frontend_means_test_specialist_provider.feature [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    And I select a category from Matter Type 2
+    Then I select the "Assign manually" button
+    And I choose a provider
+    And I select 'Assign Provider'
+
+    Then the case is assigned to the Specialist Provider
+
+    Then I select the 'Sign out' link
+    And I am logged in as "TEST_SPECIALIST_PROVIDER"
+
```
[the remark this answers is not in the document]
Comment (MishNajam) [reply behave/features/cla_frontend/frontend_means_test_specialist_provider.feature]: Done this

**Review by TawneeOwl — APPROVED** [review]
LGTM

**Review by TawneeOwl — COMMENTED** [review]
Why are the behave tests failing?

**Comment by jogilder** [comment]
> @jogilder have you tried using https://pypi.org/project/chromedriver-py/
> 
> It has a method for auto locating the chromedriver so you don't have to look it up

@said-moj this method didn't seem to work.

**Review by TawneeOwl — APPROVED** [review]
LGTM

**Review by psweeting1 — APPROVED** [review]
LGTM

**Pull request merged**